### PR TITLE
Relax schema to allow article-vor image property to exist in snippet as in model

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: src/samples/article-list/v1/first-page.json
Comment (nlisgo) [inline src/samples/article-list/v1/first-page.json:127]: This is an example of a snippet that would fail currently but is now permitted so that on lax you do not need to worry about having an image property in an article-vor snippet that doesn't have a thumbnail.

File: src/model/article-vor.v4.yaml
Comment (lsh-0) [inline src/model/article-vor.v4.yaml:5]: I was having trouble on Tuesday trying to find the right spec to validate a snippet against.

It looked like I could use the full article spec to validate a snippet but I was getting validation errors on otherwise valid data. Is the full article spec what I should be using? Because I'll persevere through the validation errors and figure out what they're actually trying to tell me.

**Comment by lsh-0** [comment]
If it's ok with you I'll hang off on approving this today until I can test it with lax tomorrow. I have some new tests that would benefit from this.

File: src/model/article-vor.v4.yaml
Comment (nlisgo) [reply src/model/article-vor.v4.yaml:5]: When I validate a snippet I always do it in the context of a list, so I would wrap the snippet in an array and use https://github.com/elifesciences/api-raml/blob/develop/dist/model/article-list.v1.json

**Comment by nlisgo** [comment]
> If it's ok with you I'll hang off on approving this today until I can test it with lax tomorrow. I have some new tests that would benefit from this.

Sure. We are keen to re-enable the ability to add interesting images and social images to articles but understand that we need to get this right. Awaiting any feedback.

File: src/model/article-vor.v4.yaml
Comment (lsh-0) [reply src/model/article-vor.v4.yaml:5]: That's clever, I thought about doing something similar but figured I was just looking in the wrong place.

My tests are working with this now but I wonder what it would take to tease the article snippet out into it's own schema - or if it's even worth it.

Comment (nlisgo) [reply src/model/article-vor.v4.yaml:5]: > My tests are working with this now but I wonder what it would take to tease the article snippet out into it's own schema - or if it's even worth it.

You could do it, it would require another script similar to `./compile.js` targeting the `./src/snippets` folder. Personally, I don't think it's worth it. `article-list.v1.json` is sufficient.

**Comment by lsh-0** [comment]
so, if I'm reading this change correctly, you've moved the contents of `common/article-vor.v1.yaml` into `snippets/article-vor.v1.yaml` and dropped the `image: required: thumbnail` constraint in favour of either a 'thumbnail' or a 'social' or both - and both are just regular images.

Here are two examples of fragments in lax right now:

```json
{
    "image": {
        "social": {
            "uri": "https://iiif.elifesciences.org/journal-cms/article%2Fsocial%2F2020-11%2F58984-b-carotene.jpg",
            "alt": "",
            "source": {
                "mediaType": "image/jpeg",
                "uri": "https://iiif.elifesciences.org/journal-cms/article%2Fsocial%2F2020-11%2F58984-b-carotene.jpg/full/full/0/default.jpg",
                "filename": "58984-b-carotene.jpg"
            },
            "size": {
                "width": 1487,
                "height": 1058
            }
        }
    }
}
---
{
    "image": {
        "thumbnail": {
            "uri": "https://iiif.elifesciences.org/journal-cms/article%2F2020-07%2Farabidopsis-illo.jpg",
            "alt": "Arabidopsis illustration",
            "source": {
                "mediaType": "image/jpeg",
                "uri": "https://iiif.elifesciences.org/journal-cms/article%2F2020-07%2Farabidopsis-illo.jpg/full/full/0/default.jpg",
                "filename": "arabidopsis-illo.jpg"
            },
            "size": {
                "width": 1000,
                "height": 500
            }
        }
    }
}
```

After being merged in it would add or overwrite the image at `image.thumbnail` or `image.social` with the above.

Given how lax behaves by just taking a set of top-level keys from an article to create a snippet, I *think* this change should be ok. It's a tentative approval though.

**Review by lsh-0 — APPROVED** [review]

File: src/model/article-vor.v4.yaml
Comment (nlisgo) [inline src/model/article-vor.v4.yaml:5]: @lsh-0 this is the significant change. The `article-vor.v4` schema not inherits from the whole of the snippet schema. So a valid full article will produce a valid snippet.

Comment (lsh-0) [reply src/model/article-vor.v4.yaml:5]: cheers, @nlisgo 

feel free to merge

**Comment by nlisgo** [comment]
@lsh-0 I need to hold off. I had overlooked adding support for social image in the article poa snippet. I won't merge this in until I have a green run over here: https://github.com/elifesciences/api-sdk-php/pull/216

**Comment by nlisgo** [comment]
2 green downstream PR's:

- https://github.com/elifesciences/api-dummy/pull/174
- https://github.com/elifesciences/api-sdk-php/pull/216

This is ready to roll!

**Comment by lsh-0** [comment]
Brave man updating api-raml on a Friday ;) Good luck. 

cc @gnott there should be some bot-lax and lax PRs popping up over this. It shouldn't affect either until I add snippet merging and validation back on but let's hold off merging them just in case.

**Comment by gnott** [comment]
Yes, PR https://github.com/elifesciences/bot-lax-adaptor/pull/703, I'll ignore it and leave it for you @lsh-0 please.

**Comment by lsh-0** [comment]
cheers, @gnott 

update: merged

**Comment by lsh-0** [comment]
hrm ... kinda expected a PR for lax after merging that box-lax PR. 

download-s

**Pull request merged**
